java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessageSender.java

```diff
@@ -521,8 +566,9 @@ public void onSendComplete(final Delivery delivery)
 			{
 				this.lastKnownLinkError = null;
 				this.retryPolicy.resetRetryCount(this.getClientId());
-				pendingSend.complete(null);
+
 				pendingSendWorkItem.getTimeoutTask().cancel(false);
+				pendingSend.complete(null);
```

Comment:
Can we just create a method like

public void completePendingSend(CompletableFuture<void> pendingSend, Exception exception) 

which do these patterns in one place?